bentoml/frameworks/pytorch.py

```diff
@@ -114,3 +131,119 @@ def get(self):
 
     def save(self, dst):
         return cloudpickle.dump(self._model, open(self._file_path(dst), "wb"))
+
+
+class PytorchLightningModelArtifact(BentoServiceArtifact):
+    """Abstraction for saving and loading pytorch lightning model
+
+    Args:
+        name (string): Name of the pytorch model
+    Raises:
+        MissingDependencyException: torch and pytorch_lightning package is required.
+
+    Example usage:
+
+    >>>
+    >>> # Train pytorch lightning model
+    >>> from pytorch_lightning.core.lightning import LightningModule
+    >>>
+    >>> class SimpleModel(LightningModule):
+    >>>     def forward(self, x):
+    >>>         return x.add(1)
+    >>>
+    >>> model = SimpleModel()
+    >>>
+    >>> import bentoml
+    >>> import torch
+    >>> from bentoml.adapters import JsonInput
+    >>> from bentoml.frameworks.pytorch import PytorchLightningModelArtifact
+    >>>
+    >>> @bentoml.env(infer_pip_packages=True)
+    >>> @bentoml.artifacts([PytorchLightningModelArtifact('model')])
+    >>> class PytorchLightingService(bentoml.BentoService):
+    >>>     @bentoml.api(input=DataframeInput(), batch=True)
+    >>>     def predict(self, df):
+    >>>         input_tensor = torch.from_numpy(df.to_numpy())
+    >>>         return self.artifacts.model(input).numpy()
+    >>>
+    >>> svc = PytorchLightingService()
+    >>> # Pack Pytorch Lightning model instance
+    >>> svc.pack('model', model)
+    >>>
+    >>> # Pack saved Pytorch Lightning model
+    >>> # import torch
+    >>> # script = model.to_torchscript()
+    >>> # saved_model_path = 'path/to/model.pt'
+    >>> # torch.jit.save(script, saved_model_path)
+    >>> # svc.pack('model', saved_model_path)
+    >>>
+    >>> svc.save()
+    """
+
+    def __init__(self, name):
+        super(PytorchLightningModelArtifact, self).__init__(name)
+        self._model = None
+        self._model_path = None
+
+    def _saved_model_file_path(self, base_path):
+        return os.path.join(base_path, self.name + '.pt')
+
+    def pack(self, path_or_model, metadata=None):  # pylint:disable=arguments-differ
+        if _is_pytorch_lightning_model_file_like(path_or_model):
+            logger.info('PytorchLightningArtifact is packing a saved model from path')
+            self._model_path = path_or_model
+        else:
+            try:
+                from pytorch_lightning.core.lightning import LightningModule
+            except ImportError:
+                raise InvalidArgument(
+                    '"pytorch_lightning.lightning.LightningModule" model is required '
+                    'to pack a PytorchLightningModelArtifact'
+                )
+            if isinstance(path_or_model, LightningModule):
+                logger.info(
+                    'PytorchLightningArtifact is packing a pytorch lightning '
+                    'model instance'
+                )
+                self._model = path_or_model
```

Comment:
Is there any behaviour gap between the loaded model and the original?

Reply:
The original is LightningModule and the loaded one is torchscript.

There should be behavior differences between them. I am not sure how much its potential affects. We are using the official recommended way for inference in production https://pytorch-lightning.readthedocs.io/en/stable/production_inference.html

Reply:
Then I think we should save&load once `pack` called and replace the `_model` so that users could get a consistent behavior of `svc.artifact.model`.